From 7285a626a3fd36433ca2dec1d6b5505a2d43e28d Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Mon, 3 Dec 2012 18:00:43 +0100
Subject: [PORT FROM MAIN]wifi: multi-vendor enabling in mainline

BZ: 60525 65640

Adapt netd so that it can fit both TI and BCM solutions.
In order to achieve that, this patch create 2 netd versions,
one for TI an other for BCM.

Feature   : Wlan
Category  : Integration
Purpose   : ABSP-Adapter
Origin    : Upstream
Upstream-candidate: no

Change-Id: I0352b486f5320f6152af4b49a6b72ef20854f1ba
Orig-Change-Id: Ia2eb833e7a1afd2a3ecbe84c58c4e6491d6b900f
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
Reviewed-on: http://android.intel.com:8080/78501
Reviewed-by: Naoumenko, PaulX <paulx.naoumenko@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Jeremie GARCIA <jeremie.garcia@intel.com>
---
 Android.mk |   77 +++++++++++++++++++++++++++++++++++++++++++++++++-----------
 1 files changed, 63 insertions(+), 14 deletions(-)

diff --git a/Android.mk b/Android.mk
index 93c5d1a..482ba6a 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,6 +1,13 @@
 LOCAL_PATH:= $(call my-dir)
 
+########################################
+ifeq ($(strip $(BOARD_WLAN_DEVICE)),wl12xx-compat)
+########################################
+
 include $(CLEAR_VARS)
+LOCAL_MODULE:= netd.ti
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_STEM := netd
 
 LOCAL_SRC_FILES:=                                      \
                   BandwidthController.cpp              \
@@ -23,14 +30,60 @@ LOCAL_SRC_FILES:=                                      \
                   oem_iptables_hook.cpp                \
                   UidMarkMap.cpp                       \
                   main.cpp                             \
+LOCAL_SRC_FILES += SoftapControllerTI.cpp
+
+LOCAL_C_INCLUDES := $(KERNEL_HEADERS) \
+                    external/mdnsresponder/mDNSShared \
+                    external/openssl/include \
+                    external/stlport/stlport \
+                    bionic \
+                    bionic/libc/private \
+                    $(call include-path-for, libhardware_legacy)/hardware_legacy \
+
+LOCAL_C_INCLUDES += external/libnl-headers
+LOCAL_CFLAGS += -DCONFIG_LIBNL20 -Werror=format
+
 
-ifeq ($(BOARD_HAVE_TI12XX),true)
-   LOCAL_SRC_FILES += SoftapControllerTI.cpp
+
+LOCAL_SHARED_LIBRARIES := libstlport libsysutils liblog libcutils libnetutils \
+                          libcrypto libhardware_legacy libmdnssd libdl
+
+LOCAL_STATIC_LIBRARIES := libnl_2
+
+include $(BUILD_EXECUTABLE)
+
+########################################
 else
-   LOCAL_SRC_FILES += SoftapController.cpp
-endif
+########################################
+
+include $(CLEAR_VARS)
+LOCAL_MODULE:= netd.bcm
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_STEM := netd
+LOCAL_CFLAGS :=
+LOCAL_SRC_FILES:=                                      \
+                  BandwidthController.cpp              \
+                  CommandListener.cpp                  \
+                  DnsProxyListener.cpp                 \
+                  FirewallController.cpp               \
+                  IdletimerController.cpp              \
+                  InterfaceController.cpp              \
+                  MDnsSdListener.cpp                   \
+                  NatController.cpp                    \
+                  NetdCommand.cpp                      \
+                  NetdConstants.cpp                    \
+                  NetlinkHandler.cpp                   \
+                  NetlinkManager.cpp                   \
+                  PppController.cpp                    \
+                  ResolverController.cpp               \
+                  SecondaryTableController.cpp         \
+                  TetherController.cpp                 \
+                  ThrottleController.cpp               \
+                  oem_iptables_hook.cpp                \
+                  logwrapper.c                         \
+                  main.cpp                             \
 
-LOCAL_MODULE:= netd
+LOCAL_SRC_FILES += SoftapController.cpp
 
 LOCAL_C_INCLUDES := $(KERNEL_HEADERS) \
                     external/mdnsresponder/mDNSShared \
@@ -40,23 +93,19 @@ LOCAL_C_INCLUDES := $(KERNEL_HEADERS) \
                     bionic/libc/private \
                     $(call include-path-for, libhardware_legacy)/hardware_legacy \
 
-ifeq ($(BOARD_HAVE_TI12XX),true)
-   LOCAL_C_INCLUDES += external/libnl-headers
-   LOCAL_CFLAGS := -DCONFIG_LIBNL20 -Werror=format
-else
-   LOCAL_CFLAGS := -Werror=format
-endif
+LOCAL_CFLAGS += -Werror=format
 
 LOCAL_SHARED_LIBRARIES := libstlport libsysutils liblog libcutils libnetutils \
                           libcrypto libhardware_legacy libmdnssd libdl \
                           liblogwrap
 
-ifeq ($(BOARD_HAVE_TI12XX),true)
-   LOCAL_STATIC_LIBRARIES := libnl_2
-endif
 
 include $(BUILD_EXECUTABLE)
 
+########################################
+endif
+########################################
+
 include $(CLEAR_VARS)
 LOCAL_SRC_FILES:=          \
                   ndc.c \
-- 
1.7.4.1

